src/sentry/integrations/slack/tasks.py

```diff
@@ -0,0 +1,128 @@
+from sentry.models import Integration, Project, Rule
+from sentry.integrations.slack.utils import get_channel_id_with_timeout
+
+from sentry.utils.redis import redis_clusters
+from sentry.mediators import project_rules
+
+MEMBER_PREFIX = "@"
+CHANNEL_PREFIX = "#"
+
+LIST_TYPES = [
+    ("channels", "channels", CHANNEL_PREFIX),
+    ("groups", "groups", CHANNEL_PREFIX),
+    ("users", "members", MEMBER_PREFIX),
+]
+
+
+class RedisRuleStatus(object):
+    def __init__(self, uuid=None):
+        self._uuid = uuid or self._generate_uuid()
+
+        cluster_id = getattr(settings, "SENTRY_RULE_TASK_REDIS_CLUSTER", "default")
+        self.client = redis_clusters.get(cluster_id)
+        self._set_inital_value()
+
+    @property
+    def uuid(self):
+        return self._uuid
+
+    def set_value(self, status, rule_id=None):
+        value = self._format_value(status, rule_id)
+        self.client.set(self._get_redis_key(), u"{}".format(value), ex=60 * 60)
+
+    def get_value(self):
+        key = self._get_redis_key()
+        value = self.client.get(key)
+        return json.loads(value)
+
+    def _generate_uuid(self):
+        return uuid4().hex
+
+    def _set_inital_value(self):
+        value = json.dumps({"status": "pending"})
+        self.client.set(self._get_redis_key(), u"{}".format(value), ex=60 * 60, nx=True)
+
+    def _get_redis_key(self):
+        return u"slack-channel-task:1:{}".format(self.uuid)
+
+    def _format_value(self, status, rule_id):
+        value = {"status": status}
+        if rule_id:
+            value["rule_id"] = six.text_type(rule_id)
+        if status == "failed":
+            value[
+                "error"
+            ] = "The slack resource does not exist or has not been granted access in that workspace."
+
+        return json.dumps(value)
+
+
+@instrumented_task(name="sentry.integrations.slack.search_channel_id", queue="integrations")
+def find_channel_id_for_rule(
+    name, environment, project, action_match, conditions, actions, frequency, uuid, rule_id=None
+):
+    redis_rule_status = RedisRuleStatus(uuid)
+
+    try:
+        project = Project.objects.get(id=project.id)
+    except Project.DoesNotExist:
+        redis_rule_status.set_value("failed")
+        return
+
+    organization = project.organization
+    integration_id = None
+    channel_name = None
+
+    for action in actions:
+        if action.get("workspace") and action.get("channel"):
+            integration_id = action["workspace"]
+            # we need to strip the prefix when searching on the channel name
+            channel_name = action["channel"].strip("#@")
+            break
+
+    try:
+        integration = Integration.objects.get(
+            provider="slack", organizations=organization, id=integration_id
+        )
+    except Integration.DoesNotExist:
+        redis_rule_status.set_value("failed")
+        return
+
+    (prefix, item_id, _timed_out) = get_channel_id_with_timeout(integration, channel_name, 3 * 60)
```

Comment:
@MeredithAnya Awesome that you could incorporate my changes :D 

Could we get a comment on the value of `3 * 60`? I think it deserves some explanation.